dcs/coalition.py

```diff
@@ -257,6 +257,13 @@ def load_from_dict(self, mission, d) -> List[StatusMessage]:
                         static_group.add_unit(static)
                     _country.add_static_group(static_group)
             self.add_country(_country)
+
+        # iterate over all .miz countries in coalition, even without any units
+        # on the map, and add them to the respective coalition
+        for country_id in countries_in_coalition.values():
+            if self.country_by_id(country_id) is None:
```

Comment:
This lookup is wasteful. If you delete L259 you don't need it, right? (that was what I'd tried to suggest in my earlier comment, sorry for not being more clear)

Reply:
@DanAlbert - I don't think so or I am not seeing something. As I understand d["country"] is a dict of countries with units on the map. The block which includes L259 goes through all countries with units on the map and creates the respective country objects (and their units) and adds them to the coalition. The block I added goes over the mission's coalitions dict, which is essentially a list of country_ids belonging to coalition - and does not have any unit information. So add_country would be called twice for countries already processed, effectively deleting all their units since countries.get_by_id() creates a new object. Right? I'll update the mission file to have 1 unit in one of the countries and test to validate that loaded and saved missions both contain the same amount of units. Let me know if you see it differently.

Reply:
Oh, I see what's going on. This is a little odd but it's good enough. I don't expect this is ever going to be a performance issue, but I thought the optimization was trivial rather than a minor refactor.

Reply:
@DanAlbert - agreed, initial patch had it in a single loop, but I think your suggestion to keep it outside that loop makes more readable since it's focusing on just countries without units on the map.